src/PlatformEngines.jl

```diff
@@ -16,11 +16,22 @@ const exe7z = Ref("")
 function __init__()
     exe7z[] = ""
     name = "7z"
-    Sys.iswindows() && (name = "$name.exe")
-    for dir in (joinpath("..", "libexec"), ".")
-        path = normpath(Sys.BINDIR, dir, name)
-        isfile(path) || continue
-        exe7z[] = path
+    if Sys.iswindows()
```

Comment:
Tangential question, but what is the reason for doing this in `init` instead of lazily at first call?

Reply:
Probably to avoid the pain point that we had with `probe_platform_engines!()`; when other tooling wanted to use the stuff in PlatformEngines but didn't want to call the same functions that would lazily-initialize things, we'd only notice that something would break when that function would be called after not calling a `Pkg.add()` operation first, or similar.

I think in this case would be simple enough to add a `get_7z_exe()` function or somesuch though.

Reply:
Yeah, I didn't mean to have a dedicated function to initialize. Just initialize when it is called the first time. I can PR that, np.

Reply:
My suggestion would be to have an `exe7z()` function that does this logic the first time it's called and then saves the path in a global and returns it from there on. Then we can just call `exe7z()` wherever we use this.

Reply:
https://github.com/JuliaLang/Pkg.jl/pull/2204 maybe. Made towards this branch. cc @staticfloat